src/mono/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -1826,11 +1823,11 @@ public static FieldInfo GetField(Type type, FieldInfo field)
                 throw new ArgumentException(SR.Argument_InvalidFieldDeclaringType, nameof(type));
 
             if (field is FieldOnTypeBuilderInst)
-                throw new ArgumentException("The specified field must be declared on a generic type definition.", nameof(field));
+                throw new ArgumentException(SR.Argument_FieldNeedGenericDeclaringType, nameof(field));
 
             FieldInfo res = type.GetField(field);
             if (res == null)
```

Comment:
Change this to assert

Reply:
This comment was meant for `if (res == null)`, not for `if (field is FieldOnTypeBuilderInst)` line.